fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/ProgressiveLoanInterestRefundServiceImpl.java

```diff
@@ -81,6 +85,13 @@ private Money recalculateTotalInterest(AdvancedPaymentScheduleTransactionProcess
         List<LoanRepaymentScheduleInstallment> installmentsToReprocess = new ArrayList<>(
                 loan.getRepaymentScheduleInstallments().stream().filter(i -> !i.isReAged() && !i.isAdditional()).toList());
 
+        if (loan.getLoanProductRelatedDetail() != null
```

Comment:
Same logic can applied here like in `ProgressiveLoanSummaryDataProvider.java:computeTotalUnpaidPayableNotDueInterestAmountOnActualPeriod`

If the loan is charged-off we dont need to recalculate interest, rather just simply returning the total interest outstanding of the loan!